conans/client/build/msbuild.py

```diff
@@ -1,21 +1,116 @@
+import re
+
 from conans import tools
-from conans.client.build.visual_environment import VisualStudioBuildEnvironment
-from conans.client.tools.win import msvc_build_command
+from conans.client.build.visual_environment import VisualStudioBuildEnvironment, vs_build_type_flags, vs_std_cpp
+from conans.client.tools.oss import cpu_count
+from conans.client.tools.win import vcvars_command
+from conans.errors import ConanException
+from conans.util.env_reader import get_env
 
 
 class MSBuild(object):
 
     def __init__(self, conanfile):
-        self._conanfile = conanfile
-        self.build_env = VisualStudioBuildEnvironment(self._conanfile)
+        if conanfile:  # FIXME: Remove when deprecated win.py msbuild tools
+            self._conanfile = conanfile
+            self._settings = self._conanfile.settings
+            self._output = self._conanfile.output
+            self.build_env = VisualStudioBuildEnvironment(self._conanfile)
 
     def build(self, project_file, targets=None, upgrade_project=True, build_type=None, arch=None,
               parallel=True, force_vcvars=False, toolset=None, platforms=None):
         with tools.environment_append(self.build_env.vars):
-            command = msvc_build_command(self._conanfile.settings,
-                                         project_file, targets=targets,
-                                         upgrade_project=upgrade_project,
-                                         build_type=build_type, arch=arch, parallel=parallel,
-                                         force_vcvars=force_vcvars, toolset=toolset,
-                                         platforms=platforms)
-            return self._conanfile.run(command)
+            # Path for custom properties file
+            props_file_contents = self._get_props_file_contents()
+            with tools.tmp_file(props_file_contents) as props_file_path:
+                vcvars = vcvars_command(self._conanfile.settings, force=force_vcvars)
+                command = self.get_command(project_file, props_file_path,
```

Comment:
I ignore how this will work with the local flow.
In the local flow, it is expected that users will just open the VS .sln, not call ``conan build``, so this will not be injected.

On the other hand, the external dependencies .props file has to be always added when there are dependencies, in the solution (not injected in  build()).

Reply:
I don't understand.